### ISSUE TITLE
Permission does not apply and give error 500

### ISSUE THREAD

**Issue by alirezastack** [metabase/metabase#5394]
When I want to set permissions for groups and revoke permissions from all users in UI nothing happens, when I open firebug (developer tools) in network (Net) tab I see 500 server error with the below response:

```
Request URL:http://159.100.248.41:3000/api/permissions/graph
Request Method:PUT
Status Code:500 Server Error
```

Request payload:
`{"revision":4,"groups":{"1":{"1":{"native":"write","schemas":"all"},"2":{"native":"read","schemas":"none"},"34":{"native":"none","schemas":"none"},"66":{"native":"none","schemas":"none"}},"2":{"1":{"native":"write","schemas":"all"},"2":{"native":"write","schemas":"all"},"34":{"native":"write","schemas":"all"},"66":{"native":"write","schemas":"all"}},"3":{"1":{"native":"write","schemas":"all"},"2":{"native":"write","schemas":"all"},"34":{"native":"write","schemas":"all"},"66":{"native":"write","schemas":"all"}},"4":{"1":{"native":"none","schemas":"none"},"2":{"native":"none","schemas":{"":{"5":"none","6":"all","7":"all","8":"all","9":"none"}}},"34":{"native":"none","schemas":"none"},"66":{"native":"none","schemas":"none"}},"5":{"1":{"native":"none","schemas":"none"},"2":{"native":"none","schemas":{"":{"5":"none","6":"all","7":"all","8":"all","9":"all"}}},"34":{"native":"none","schemas":"none"},"66":{"native":"none","schemas":"none"}}}}`

The response:
```
{"type":"schema.core/error","schema":[{"schema":{"revision":{"p?":"clojure.core$integer_QMARK_@58475d48"
,"pred-name":"integer?"},"groups":{"schema.core.Constrained@1d55f419":{"schema.core.Constrained@1d55f419"
:{"schema":{"schema.core.OptionalKey@80fab393":{"vs":["read","write","none"]},"schema.core.OptionalKey
@e0ca6a71":{"schemas":[{"vs":["all","none"]},{"class java.lang.String":{"schemas":[{"vs":["all","none"
]},{"schema.core.Constrained@1d55f419":{"vs":["all","none"]}}]}}]}},"postcondition":"metabase.models
.permissions$check_native_and_schemas_permissions_allowed_together@4079e162","post-name":"DB permissions
 with a valid combination of values for :native and :schemas"}}}},"optional?":false,"name":"new-graph"
}],"value":[{"revision":4,"groups":{"1":{"1":{"native":"write","schemas":"all"},"2":{"native":"read"
,"schemas":"none"},"34":{"native":"none","schemas":"all"},"66":{"native":"write","schemas":"all"}},"2"
:{"1":{"native":"write","schemas":"all"},"2":{"native":"write","schemas":"all"},"34":{"native":"write"
,"schemas":"all"},"66":{"native":"write","schemas":"all"}},"3":{"1":{"native":"write","schemas":"all"
},"2":{"native":"write","schemas":"all"},"34":{"native":"write","schemas":"all"},"66":{"native":"write"
,"schemas":"all"}},"4":{"1":{"native":"none","schemas":"none"},"2":{"native":"none","schemas":{"":{"5"
:"none","6":"all","7":"all","8":"all","9":"none"}}},"34":{"native":"none","schemas":"none"},"66":{"native"
:"none","schemas":"none"}},"5":{"1":{"native":"none","schemas":"none"},"2":{"native":"none","schemas"
:{"":{"5":"none","6":"all","7":"all","8":"all","9":"all"}}},"34":{"native":"none","schemas":"none"},"66"
:{"native":"none","schemas":"none"}}}}],"error":["schema.utils.NamedError@223b6f0e"],"message":"Input
 to update-graph! does not match schema: \n\n\t \u001B[0;33m  [(named {:groups {1 {2 (not (\"DB permissions
 with a valid combination of values for :native and :schemas\" a-clojure.lang.PersistentArrayMap))}}
} new-graph)] \u001B[0m \n\n","stacktrace":["models.permissions$fn__20038$update_graph_BANG___20047.invoke
(permissions.clj:477)","api.permissions$fn__37718$fn__37720.invoke(permissions.clj:73)","api.common.internal$do_with_caught_api_exceptions
.invokeStatic(internal.clj:229)","api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj
:224)","api.permissions$fn__37718.invokeStatic(permissions.clj:61)","api.permissions$fn__37718.invoke
(permissions.clj:61)","middleware$enforce_authentication$fn__38784.invoke(middleware.clj:120)","api.routes$fn__38928
.invokeStatic(routes.clj:68)","api.routes$fn__38928.invoke(routes.clj:68)","routes$fn__39540$fn__39541
.doInvoke(routes.clj:64)","routes$fn__39540.invokeStatic(routes.clj:60)","routes$fn__39540.invoke(routes
.clj:60)","middleware$log_api_call$fn__38883$fn__38885.invoke(middleware.clj:329)","middleware$log_api_call$fn__38883
.invoke(middleware.clj:328)","middleware$add_security_headers$fn__38833.invoke(middleware.clj:243)","middleware$bind_current_user$fn__38788
.invoke(middleware.clj:140)","middleware$maybe_set_site_url$fn__38837.invoke(middleware.clj:266)"]}
```
I have used different browsers (Chrome, Firefox) to check whether it is related to browsers or not. What is the reason for this? I think this might be a bug on Metabase side, could someone shed some light on the issue?

**Comment by alirezastack** [comment]
I'm using Metabase 0.24.1

**Comment by attekei** [comment]
@alirezastack Okay, then I think that updating Metabase should solve your problem! Based on the request payload you are having the exactly same issue that was fixed in #5178.
